packages/frontend/src/hooks/useCrabPosition/useCrabPosition.ts

```diff
@@ -87,6 +87,84 @@ export const useCrabPosition = (user: string) => {
   }
 }
 
+/*
+  depositedEth = Sum of deposited ethAmount - Sum of withdrawn ethAmount
+  depositedUsd = Sum of deposited ethUsed - Sum of withdrawn ethUsd
+  minCurrentEth = currentEth 
+  minCurrentUsd = currentEth * indexPrice
+  minPnlUsd = minCurrentUsd - depositedUsd
+  minPnL = minPnlUsd / depositedUsd * 100
+*/
+export const useCrabPositionV2 = (user: string) => {
+  const crabLoading = useAtomValue(crabLoadingAtomV2)
+  const isCrabPositionValueLoading = useAtomValue(crabPositionValueLoadingAtomV2)
+  const currentEthValue = useAtomValue(currentCrabPositionValueInETHAtomV2)
+
+  const { loading: txHistoryLoading, data: txHistoryData } = useUserCrabTxHistory(user)
+
+  const index = useAtomValue(indexAtom)
+  const ethIndexPrice = toTokenAmount(index, 18).sqrt()
+
+  const [minCurrentEth, setMinCurrentEth] = useState(BIG_ZERO)
+  const [minCurrentUsd, setMinCurrentUsd] = useState(BIG_ZERO)
+  const [minPnlUsd, setMinPnlUsd] = useState(BIG_ZERO)
+  const [minPnL, setMinPnL] = useState(BIG_ZERO)
+
+  const { depositedEth, usdAmount: depositedUsd } = useAppMemo(() => {
```

Comment:
This seems good with one exception. Migrated users will not have a deposit tx. so this PnL / position might be wrong for users. We should track migration events too and consider them as deposits.  @alexisgauba What do you think will be the right thing here?